### PR TITLE
DRILL-4359: Adds equals/hashCode methods to EndpointAffinity

### PR DESCRIPTION
**Pull request by laurentgo**
Adds equals/hashCode methods to EndpointAffinity to allow for comparison in tests.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/physical/EndpointAffinity.java [hunk 1]
```diff
@@ -96,6 +96,42 @@ public boolean isAssignmentRequired() {
...
+    }
+    EndpointAffinity other = (EndpointAffinity) obj;
+    if (Double.doubleToLongBits(affinity) != Double.doubleToLongBits(other.affinity)) {
+      return false;
+    }
+    if (endpoint == null) {
+      if (other.endpoint != null) {
+        return false;
+      }
+    } else if (!endpoint.equals(other.endpoint)) {
```
Comment (jaltekruse) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/EndpointAffinity.java]: It looks like DrillbitEndpoint also lacks equals and hashcode methods, are these going to be necessary for your tests as well?

Comment (laurentgo) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/EndpointAffinity.java]: DrillbitEndpoint is a protobuf Message, so the equals/hashCode methods are provided by protobuf API (by AbstractMessage if I'm correct)

Comment (jaltekruse) [inline exec/java-exec/src/main/java/org/apache/drill/exec/physical/EndpointAffinity.java]: I should have looked more carefully, I was looking at the one in the proto.beans package, but that isn't being used here.

**Comment by jaltekruse** [comment]
+1

**Pull request closed without merging**